### ISSUE TITLE
Unable to find TreeViewItem until clicked manually

### ISSUE THREAD

**Issue by Xiphereal** [microsoft/WinAppDriver#1586]
Hi,
I'm using WinAppDriver driven by Appium, trying to test a WPF application.
I'm trying to perform a double click (#1584) on a `TreeViewItem`, but the problem is that the item cannot be found by WinAppDriver in any way until it is first clicked and hovered manually. When inspecting the app with Inspect.exe the same happens, being only display 2 elements of the `TreeView`, and when the `TreeViewItem` is clicked and hovered, then all 8 `TreeViewItem` are displayed as expected.

Any idea why this happens? Any workaround?

**Comment by anunay1** [comment]
Can you hover over the element with Mouse.move()

**Comment by Xiphereal** [comment]
> 
> 
> Can you hover over the element with Mouse.move()

No, I can't move the mouse over the element since I can't find the element. Which method do you refer exactly with `Mouse.move()`?

Also, it seems that **the element needs to be both hovered and clicked to be visible** for WinAppDriver. It seems to be some issue with the "automation visibility" of the `TreeViewItem`.

I guess the last resort would be to perform the double click on a certain viewport or screen `Coordinates`, but this is the least reliable method to perform an action over an element, being flaky and broken many times :(

**Comment by anunay1** [comment]
Which language binding are you using?

**Comment by Xiphereal** [comment]
> 
> 
> Which language binding are you using?

You mean the client, right? It's C#, the dotNET client.

**Comment by anunay1** [comment]
Yes, the client, can you try to use keyboard commands as a work around.

**Comment by anunay1** [comment]
or else you can use findelementbyImage.

**Comment by anunay1** [comment]
> > Can you hover over the element with Mouse.move()
> 
> No, I can't move the mouse over the element since I can't find the element. Which method do you refer exactly with `Mouse.move()`?
> 
> Also, it seems that **the element needs to be both hovered and clicked to be visible** for WinAppDriver. It seems to be some issue with the "automation visibility" of the `TreeViewItem`.
> 
> I guess the last resort would be to perform the double click on a certain viewport or screen `Coordinates`, but this is the least reliable method to perform an action over an element, being flaky and broken many times :(

Yes I agree its flaky and not a good solution.

**Comment by anunay1** [comment]
> or else you can use findelementbyImage but that only supports clicking as of now, but will solve your problem.

@Xiphereal Did this solve the problem?

**Comment by Xiphereal** [comment]
I haven't had the time yet to look it up. I'll keep you updated :).
